### PR TITLE
Add Scale3D component

### PR DESCRIPTION
**Pull request by Wumpf**
### What

* Part of #﻿6831
* Based on https://github.com/rerun-io/rerun/pull/6866

Introduces a new `Scale3D` component that lives directly on the `Transform3D` archetype.
The `Scale3D` _datatype_ which is part of `TranslationRotationScale3D` is still around, but will be removed in a subsequent PR.

Additionally, reversed order of how transform components. Rationale: "translation, rotation, scale" is a common way of expressing a simple transform. What is meant is that we first scale an object, then rotate it and _then_ translate it.

Commit by commit review possible.

### Checklist
* [x] pass `main` ci
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6892?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6892?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!
* [x] If have noted any breaking changes to the log API in `CHANGELOG.md` and the migration guide

- [PR Build Summary](https://build.rerun.io/pr/6892)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

File: crates/viewer/re_space_view_spatial/src/contexts/transform_context.rs [hunk 1]
```diff
@@ -312,18 +312,23 @@ fn get_parent_from_child_transform(
...
             Translation3D::name(),
+            Scale3D::name(),
+            TransformMat3x3::name(),
         ],
     );
     if result.components.is_empty() {
         return None;
     }
 
+    // Order is specified by order of components in the Transform3D archetype.
```
Comment (emilk) [inline crates/viewer/re_space_view_spatial/src/contexts/transform_context.rs:324]: We could add a `debug_assert` here to check that the fields in the archetype matches what we expect by the code below, by using the reflection API (`ArchetypeReflection::fields`)

File: docs/content/reference/migration/migration-0-18.md [hunk 2]
```diff
@@ -89,6 +90,18 @@ Note that the order of the method calls does _not_ affect the order in which tra
 an empty archetype instead that you can populate (e.g. `rerun::Transform3D().with_mat3x3(rerun::datatypes::Mat3x3::IDENTITY)`).
 
 
+Scale is no longer an enum datatype but a component with a 3D vec:
+Before:
+```rust
+let scale_uniform = rerun::Scale3D::Uniform(2.0);
+let scale_y = rerun::Scale3D::ThreeD([1.0, 2.0, 1.0]);
```
Comment (emilk) [inline docs/content/reference/migration/migration-0-18.md:97]: We could keep `Uniform` and `ThreeD` around as functions, with `#[deprecated]` and `#[allow(non_snake_case_identifiers)]` to help the user

File: rerun_cpp/src/rerun/archetypes/transform3d_ext.cpp [hunk 3]
```diff
@@ -121,14 +121,15 @@ namespace rerun {
         ///
         /// \param _translation \copydoc Transform3D::translation
         /// \param rotation \copydoc datatypes::TranslationRotationScale3D::rotation
-        /// \param scale \copydoc datatypes::TranslationRotationScale3D::scale
+        /// \param _scale \copydoc Transform3D::scale
         /// \param from_parent \copydoc datatypes::TranslationRotationScale3D::from_parent
         Transform3D(
             const components::Translation3D& _translation, const datatypes::Rotation3D& rotation,
-            const datatypes::Scale3D& scale, bool from_parent = false
+            const components::Scale3D& _scale, bool from_parent = false
```
Comment (emilk) [inline rerun_cpp/src/rerun/archetypes/transform3d_ext.cpp:128]: https://github.com/rerun-io/rerun/blob/4c682edec32d9763e66d32d456bd3af5c2b7926d/CODE_STYLE.md#L152-L181


```suggestion
            const components::Scale3D& scale_, bool from_parent = false
```

here and everywhere

File: rerun_py/rerun_sdk/rerun/archetypes/transform3d_ext.py [hunk 4]
```diff
@@ -58,15 +58,19 @@ def __init__(
             if from_parent is None:
                 from_parent = False
 
+            if scale is not None and (not hasattr(scale, "__len__") or len(scale) == 1):  # type: ignore[arg-type]
+                print("reached scalar case")
+                scale = Scale3D(scale)  # type: ignore[arg-type]
+                print(scale)
```
Comment (emilk) [inline rerun_py/rerun_sdk/rerun/archetypes/transform3d_ext.py:64]:
```suggestion
```

File: rerun_py/rerun_sdk/rerun/archetypes/transform3d_ext.py [hunk 5]
```diff
@@ -58,15 +58,19 @@ def __init__(
             if from_parent is None:
                 from_parent = False
 
+            if scale is not None and (not hasattr(scale, "__len__") or len(scale) == 1):  # type: ignore[arg-type]
+                print("reached scalar case")
```
Comment (emilk) [inline rerun_py/rerun_sdk/rerun/archetypes/transform3d_ext.py:62]:
```suggestion
```

File: rerun_py/rerun_sdk/rerun/archetypes/transform3d_ext.py (hunk 4 above)
Comment (emilk) [reply rerun_py/rerun_sdk/rerun/archetypes/transform3d_ext.py:64]: Tip: suffix temporary code with `// TODO: remove` and it won't pass CI

**Review by emilk — APPROVED** [review]
This looks like it was painful

File: rerun_py/rerun_sdk/rerun/archetypes/transform3d_ext.py (hunk 4 above)
Comment (Wumpf) [reply rerun_py/rerun_sdk/rerun/archetypes/transform3d_ext.py:64]: argh, good you caught those!

**Comment by Wumpf** [comment]
@rerun-bot full-check

**Pull request merged**